bundles/CoreBundle/src/Command/Asset/GenerateChecksumCommand.php

```diff
@@ -0,0 +1,123 @@
+            ->setDescription('Re-generates checksum for specific or all assets')
+            ->addOption(
+                'id',
+                null,
+                InputOption::VALUE_REQUIRED | InputOption::VALUE_IS_ARRAY,
+                'only generate checksum for assets with this (IDs)'
+            )
+            ->addOption(
+                'parent',
+                'p',
+                InputOption::VALUE_OPTIONAL,
+                'only generate checksum for assets in this folder (ID)'
+            )
+            ->addOption(
+                'missing-only',
+                'm',
+                InputOption::VALUE_NONE,
+                'only generate checksum for assets which have no checksum yet'
+            )
+            ->addOption(
+                'path-pattern',
+                null,
+                InputOption::VALUE_OPTIONAL,
+                'only generate checksum for the given regex pattern (path + filename), example:  ^/Sample.*urban.jpg$'
+            );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $conditionVariables = [];
+        $missingOnly = $input->getOption('missing-only');
+
+        $conditions = [];
+
+        if ($input->getOption('parent')) {
+            $parent = Asset::getById((int) $input->getOption('parent'));
+            if ($parent instanceof Asset\Folder) {
+                $conditions[] = "path LIKE '" . Helper::escapeLike($parent->getRealFullPath()) . "/%'";
+            } else {
+                $this->writeError($input->getOption('parent') . ' is not a valid asset folder ID!');
+                exit;
+            }
+        }
+
+        if ($ids = $input->getOption('id')) {
+            $conditions[] = sprintf('id in (%s)', implode(',', $ids));
+        }
+
+        if ($regex = $input->getOption('path-pattern')) {
+            $conditions[] = 'CONCAT(`path`, filename) REGEXP ?';
+            $conditionVariables[] = $regex;
+        }
+
+        $list = new Asset\Listing();
+        $list->setCondition(implode(' AND ', $conditions), $conditionVariables);
+        $total = $list->getTotalCount();
+        $perLoop = 10;
+
+        for ($i = 0; $i < (ceil($total / $perLoop)); $i++) {
+            $list->setLimit($perLoop);
+            $list->setOffset($i * $perLoop);
+            $assets = $list->load();
+            foreach ($assets as $asset) {
+                try {
+                    if ($missingOnly && $asset->getCustomSetting('checksum')) {
+                        continue;
+                    }
+
+                    if ($asset->getType() === 'folder') {
+                        continue;
+                    }
+
+                    $this->output->writeln(' generating checksum for asset: ' . $asset->getRealFullPath() . ' | ' . $asset->getId());
+                    $asset->generateChecksum();
+                    $asset->save(['versionNote' => 'checksum generation']);
```

Comment:
is there a need to call save when `generateChecksum` updates the custom settings anyway?